pulsar-metadata/src/test/java/org/apache/pulsar/metadata/ZKSessionTest.java

```diff
@@ -156,39 +153,31 @@ public void testReacquireLeadershipAfterSessionLost() throws Exception {
         @Cleanup
         LeaderElection<String> le1 = coordinationService.getLeaderElection(String.class, path,
                 leaderElectionEvents::add);
-
+        // --- test manual elect
         le1.elect("value-1").join();
         assertEquals(le1.getState(), LeaderElectionState.Leading);
 
         LeaderElectionState les = leaderElectionEvents.poll(5, TimeUnit.SECONDS);
         assertEquals(les, LeaderElectionState.Leading);
-
+        // --- expire session
         zks.expireSession(((ZKMetadataStore) store).getZkSessionId());
 
         SessionEvent e = sessionEvents.poll(5, TimeUnit.SECONDS);
         assertEquals(e, SessionEvent.ConnectionLost);
 
         e = sessionEvents.poll(10, TimeUnit.SECONDS);
         assertEquals(e, SessionEvent.SessionLost);
-
-        Awaitility.await().untilAsserted(() -> {
-            assertEquals(le1.getState(), LeaderElectionState.Leading);
-        });
-
-        les = leaderElectionEvents.poll();
-        assertNull(les);
-
+        // --- test  le1 can be leader
+        Awaitility.await()
+                .untilAsserted(()-> assertEquals(le1.getState(),LeaderElectionState.Leading)); // reacquire leadership
         e = sessionEvents.poll(10, TimeUnit.SECONDS);
         assertEquals(e, SessionEvent.Reconnected);
         e = sessionEvents.poll(10, TimeUnit.SECONDS);
         assertEquals(e, SessionEvent.SessionReestablished);
-
-        Awaitility.await().untilAsserted(() -> {
-                    assertEquals(le1.getState(), LeaderElectionState.Leading);
-        });
-        les = leaderElectionEvents.poll();
-        assertNull(les);
-
+        // ---- after 2000 millis, the le1 is also can be leader.
+        Thread.sleep(2_000);
```

Comment:
why do we need this sleep ?

Reply:
Waiting for 2000 milliseconds to verify that the node can also be the leader.